### ISSUE TITLE
[Bug] React.jsx: type is invalid

### ISSUE THREAD

**Issue by 1446445040** [storybookjs/builder-vite#488]
### What version of `vite` are you using?

^3.0.0(latest)

### System info and storybook versions

System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.pnpm/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Safari: 15.6.1
  npmPackages:
    @storybook/addon-actions: ^6.5.10 => 6.5.10 
    @storybook/addon-backgrounds: ^6.5.10 => 6.5.10 
    @storybook/addon-docs: ^6.5.10 => 6.5.10 
    @storybook/addon-essentials: ^6.5.10 => 6.5.10 
    @storybook/addon-links: ^6.5.10 => 6.5.10 
    @storybook/addon-measure: ^6.5.10 => 6.5.10 
    @storybook/addon-outline: ^6.5.10 => 6.5.10 
    @storybook/builder-vite: ^0.2.2 => 0.1.36 
    @storybook/client-api: ^6.5.10 => 6.5.10 
    @storybook/client-logger: ^6.5.10 => 6.5.10 
    @storybook/core-common: ^6.5.10 => 6.5.10 
    @storybook/node-logger: ^6.5.10 => 6.5.10 
    @storybook/vue3: ^6.5.10 => 6.5.10 

### Describe the Bug

Sorry for bothering, but i just found an inevitable exception when i start my vue3 project without using any mdx file with `@storybook/builer-vite@1.0.38` or laster.
![image](https://user-images.githubusercontent.com/31957758/186842199-812d64c2-68c5-41eb-8308-ce869d70d6c9.png)the web page crashed due to this problem, which was fixed by adding some lines in main.js:
```js
async viteFinal(config) {
  // ignore react plugins
  config.plugins = config.plugins.flat().filter(p => !p.name.includes('react'))
  return config
}
```
i didn't use any mdx features. but I need to ignore the react plugins, which made me confused.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by 1446445040** [comment]
problem lines are as under:
https://github.com/storybookjs/builder-vite/pull/377#discussion_r955653786

**Comment by IanVS** [comment]
This seems to be a bug in the vite react plugin.  It is only [supposed to](https://github.com/vitejs/vite/blob/37ac91e5f680aea56ce5ca15ce1291adc3cbe05e/packages/plugin-react/src/index.ts#L193) apply to .mjs, .ts(x), and .js(x) files.  In your case, it looks like the query parameter is being used as the file extension, instead of the actual file extension of `.vue`.  They do specifically look at the query param though, so I'm not clear whether it's a bug or something unintended.  

Any chance you could create and share a minimal reproduction that we can use to troubleshoot this?

**Comment by 1446445040** [comment]
Of course. I tried to create a demo with codesandbox， but I failed， I'll try it again later with some other ways

**Comment by IanVS** [comment]
Thanks!  Sometimes it's easiest to create a new vite project with `npm create vite@latest`, and then add storybook with `npx sb@next init`.  Then when you've got it reproducing the issue, push the repo up to github.

**Comment by 1446445040** [comment]
@IanVS I've made the reproduction in this [repo](https://github.com/1446445040/storybook-bug-reproduction), you can check for more info. Just start the project with pnpm, and run `serve`, you'll find something wrong in console.
<img width="500" alt="截屏2022-08-30 11 06 57" src="https://user-images.githubusercontent.com/31957758/187339878-733161b8-4e3c-4394-a0b2-1416bdfd1f6c.png">

**Comment by emosheeep** [comment]
Is there any progress on this problem?😋

**Comment by IanVS** [comment]
No, I haven't had a chance to dig into it yet, but it's on my radar.

**Comment by emosheeep** [comment]
it’s okay, i’m not in a hurry.  For now, my project runs well by excluding the react plugin.

**Comment by emosheeep** [comment]
I saw this issue https://github.com/storybookjs/builder-vite/pull/469 closed and there was some related commit has been merged, I wonder if this issue could be resolved? Seems that this problem is also related to the compatibility and some people around encountered too.

**Comment by IanVS** [comment]
The issue is that we're using the react plugin for mdx, but that conflicts with vue files written in jsx/tsx.  I think we need to find a different way to handle mdx.  In 7.0 I think this will be solved by the upgrade to mdx2.  Unfortunately I don't think the vue2 PR you mention helped this situation.

**Comment by emosheeep** [comment]
Well, I got it. It seems that we have to make some choices on this. Looking forward to 7.0😄
